lib/marc_match_key/edition_key.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the edition portion of a GoldRush key
+  class EditionKey
+    include MarcMatchFunctions
+    attr_reader :record, :key
+
+    def initialize(record)
+      @record = record
+      @key ||= generate_key
+    end
+
+    private
+
+    def generate_key
+      f250 = record.fields('250').find { |f| f['a'] }
+      return pad_with_underscores('1', 3) if f250.nil?
+
+      subfa = clean_string(f250['a'])
+      if subfa =~ /[0-9]/
+        subfa.gsub!(/^[^0-9]*([0-9]+)[^0-9]*.*$/, '\1')
+      else
+        subfa.gsub!(/^[^a-z]*([a-z]+)[^0-9]*.*$/, '\1')
+      end
+      pad_with_underscores(subfa, 3)
+    end
+
+    def clean_string(string)
+      string = normalize_string_and_remove_accents(string.dup)
+      string.downcase!
+      edition_string_to_numbers(string)
```

Comment:
If we don't use the mutating versions of `gsub` and `downcase` subsequently, I think we don't need the `.dup`.

```suggestion
      string = normalize_string_and_remove_accents(string)
      edition_string_to_numbers(string.downcase)
```